src/test/java/org/opensearch/knn/index/AdvancedFilteringUseCasesIT.java

```diff
@@ -58,7 +58,7 @@ public class AdvancedFilteringUseCasesIT extends KNNRestTestCase {
 
     private static final String TERM_FIELD = "term";
 
-    private static final int k = 20;
```

Comment:
Can we add a test that validates this fix works?

Reply:
in #2511 it shows how to reproduce the bug, so i changed k to `NUM_DOCS / 2` as https://github.com/opensearch-project/k-NN/blob/00adf469c865309ecb65410d4ecdb1e122a009dc/src/test/java/org/opensearch/knn/index/AdvancedFilteringUseCasesIT.java#L459-L463